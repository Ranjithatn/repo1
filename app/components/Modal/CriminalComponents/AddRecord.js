import React, { Fragment } from "react";
import { connect } from "react-redux";
import { Link, withRouter } from "react-router-dom";
import Modal from "../Modal";
import Button from "../../Button/Button";
import { requestHideModal } from "../../../actions/modal";
import { requestInputFieldChanged } from "../../../actions/global";
import Label from "../../Label/Label";
import Select from "../../Select/Select";
import Input from "../../Input/Input";
import { requestSetAddnew } from "../../../actions/tenprint";
import { newCrimeTypeSelector } from "../../../selectors/updateCriminalModal";
// import "./UpdateCriminalModal.scss";

export function AddRecord({
  formatMessage,
  requestInputFieldChanged,
  requestSetAddnew,
  newCrimeType
}) {
  return (
    <div className="case-field">
      <div className="control">
        <Select
          defaultValue="Filter"
          formatMessage={formatMessage}
          id="CaseType"
          name="CrimeType"
          value={newCrimeType.CaseType}
          onChange={e => requestInputFieldChanged(e, "addCrimeType")}
          options={[
            {
              value: "Drug",
              displayName: formatMessage({
                id: "Drug"
              })
            },
            {
              value: "Murder",
              displayName: formatMessage({
                id: "Murder"
              })
            },
            {
              value: "Robbery",
              displayName: formatMessage({
                id: "Robbery"
              })
            }
          ]}
        />
      </div>
      <div className="control">
        <Input
          type="text"
          className="input"
          id="Description"
          placeholder={formatMessage({
            id: "typeSubject"
          })}
          value={newCrimeType.Description}
          onChange={e => requestInputFieldChanged(e, "addCrimeType")}
        />
      </div>
      <Button
        className="is-primary is-rounded"
        leftIcon="plus"
        // style={{ marginRight: 18 }}
        // text={formatMessage({ id: "+" })}
        onClick={e => {
          requestSetAddnew();
        }}
        disabled={!newCrimeType.CaseType || !newCrimeType.Description}
        id="upmodbackbtn"
      />
    </div>
  );
}
const mapState = state => ({
  newCrimeType: newCrimeTypeSelector(state)
});
export default connect(mapState, {
  requestInputFieldChanged,
  requestSetAddnew
})(AddRecord);
