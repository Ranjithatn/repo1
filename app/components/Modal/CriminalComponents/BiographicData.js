import React, { Fragment } from "react";
import { connect } from "react-redux";
import { Link, withRouter } from "react-router-dom";
import NumericInput from "react-numeric-input";
import Modal from "../Modal";
import Label from "../../Label/Label";
import Title from "../../Title/Title";
import Select from "../../Select/Select";
import Input from "../../Input/Input";
import HijriDatePicker from "../../HijriDatePicker";

// import "./UpdateCriminalModal.scss";
const regex = /($[a-z])|[A-Z][^A-Z]+/g;
function BiographicData({
  formatMessage,
  MatchedUserInfo,
  handleInputChange,
  data
}) {
  const handleChange = e => {
    console.log("handle click", e.target.value);
    // return e.target.value
    // window.alert(e.target.value)
  };
  let idType = "";
  let idIssueDate = "";

  if (
    MatchedUserInfo.person &&
    MatchedUserInfo.person.extendedData &&
    MatchedUserInfo.person.extendedData.length > 0
  ) {
    MatchedUserInfo.person.extendedData.map(data => {
      if (data.key === "ID Type") {
        idType = data.value;
      }
      if (data.key === "ID Issued Date") {
        idIssueDate = data.value;
      }
    });
  }

  return (

    <Fragment>
      <div className="criminal-fields">
        <Title is="5" text={formatMessage({ id: "biographicdata" })} />
        <div className="columns">
          <div className="column">
            <div className="control">
              <Label
                htmlFor="bioname"
                text={formatMessage({ id: "firstName" })}
              />
              <Input
                type="text"
                className="input"
                id="biofirstname"
                maxLength="25"
                onChange={e => {
                  handleInputChange(e, "firstName");
                }}
                placeholder={formatMessage({ id: "firstName" })}
                value={
                  data.firstName
                    ? data.firstName
                    :data.firstName===""?"": MatchedUserInfo && MatchedUserInfo.person && MatchedUserInfo.person.name
                      ? MatchedUserInfo.person.name.split(" ")[0].replace(",", '')
                      : ""
                }
              />
            </div>
          </div>
          <div className="column">
            <div className="control">
              <Label
                htmlFor="bioname"
                text={formatMessage({ id: "lastName" })}
              />
              <Input
                type="text"
                className="input"
                id="biolastname"
                maxLength="25"
                onChange={e => {
                  handleInputChange(e, "lastName");
                }}
                placeholder={formatMessage({ id: "lastName" })}
                value={
                  data.lastName
                    ? data.lastName
                    :data.lastName===""?"": MatchedUserInfo && MatchedUserInfo.person && MatchedUserInfo.person.name
                      ? MatchedUserInfo.person.name.split(" ")[1].replace(",", '')
                      : ""
                }
              />
            </div>
          </div>
          <div className="column">
            <div className="control">
              <Label
                htmlFor="bionationality"
                text={formatMessage({ id: "nationality" })}
              />
              <Input
                type="text"
                className="input"
                id="bionationality"
                maxLength="20"
                onChange={e => {
                  handleInputChange(e, "nationality");
                }}
                placeholder={formatMessage({ id: "nationality" })}
                value={
                  data.nationality
                    ? data.nationality :data.nationality===""?"": MatchedUserInfo.person !== undefined
                      ? MatchedUserInfo.person.nationality
                      : ""
                }
              />
            </div>
          </div>
        </div>
        <div className="columns">
          <div className="column">
            <div className="control">
              <Label
                htmlFor="biosamisid"
                text={formatMessage({ id: "SAMIS ID" })}
              />
              <Input
                type="text"
                className="input"
                id="biosamisid"
                maxLength="10"
                onChange={e => {
                  handleInputChange(e, "biosamisid");
                }}
                placeholder="#########"
                value={
                  data.biosamisid
                    ? data.biosamisid
                    :data.biosamisid===""?"": MatchedUserInfo.person !== undefined
                      ? MatchedUserInfo.person.samisid
                      : ""
                }

              />
            </div>
          </div>
          <div className="column">
            <div className="control">
              <Label htmlFor="biodate" text={formatMessage({ id: "dob" })} />
              <HijriDatePicker
                className="input"
                id="biodate"
                placeholder={formatMessage({ id: "dob" })}
                value={
                  MatchedUserInfo.person && MatchedUserInfo.person.dob
                    ? MatchedUserInfo.person.dob
                    : ""
                }
              />
            </div>
          </div>
          <div className="column">
            <div className="control">
              <Label
                htmlFor="biogender"
                text={formatMessage({ id: "gender" })}
              />
              <Input
                type="text"
                className="input"
                id="biogender"
                maxLength="10"
                onChange={e => {
                  handleInputChange(e, "biogender");
                }}
                placeholder={formatMessage({ id: "gender" })}
                value={
                  data.biogender
                    ? data.biogender
                    :data.biogender===""?"": MatchedUserInfo.person !== undefined
                      ? MatchedUserInfo.person.gender
                      : ""
                }

              />
            </div>
          </div>
        </div>
        <div className="columns">
          <div className="column">
            <div className="control">
              <Label
                htmlFor="bioidtype"
                text={formatMessage({ id: "idType" })}
              />
              <Select
                // defaultValue="ID Type"
                className="level-right"
                formatMessage={formatMessage}
                id="1"
                name="idType"
                options={[
                  {
                    value: idType ? idType : "Passport",
                    displayName: idType ? idType : "Passport"
                  },
                  {
                    value: "Driving",
                    displayName: "Driving Lincence"
                  },
                  {
                    value: "Citizenship",
                    displayName: "Citizenship Card"
                  }
                ]}
              />
            </div>
          </div>
          <div className="column">
            <div className="control">
              <Label
                htmlFor="bioidissuedate"
                text={formatMessage({ id: "issueDate" })}
              />
              <HijriDatePicker
                className="input"
                id="bioidissuedate"
                onChange={e => {
                  handleChange(e);
                }}
                value={idIssueDate}
              />
            </div>
          </div>
          <div className="column" />
        </div>
        <div className="columns">
          <div className="space-fill" />
        </div>
      </div>
    </Fragment>
  );
}
export default BiographicData;
