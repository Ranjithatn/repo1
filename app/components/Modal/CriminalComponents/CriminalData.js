import React, { Fragment } from "react";
import { connect } from "react-redux";
import { Link, withRouter } from "react-router-dom";

import Label from "../../Label/Label";
import Title from "../../Title/Title";
import Textarea from "../../Textarea/Textarea";
import Input from "../../Input/Input";
import HijriDatePicker from "../../HijriDatePicker";
import Select from "../../Select/Select";
import { requestShowNotification } from "../../../actions/notifications";

// import "./UpdateCriminalModal.scss";
const regex = /($[a-z])|[A-Z][^A-Z]+/g;
function CriminalaData({
  formatMessage,
  MatchedUserInfo,
  handleInputChange,
  data={},
  lookups,
  rowId,
  crimeTypes,
  requestShowNotification,
}) {

  console.log("CriminalaData::data", data);

  // console.log("CriminalaData-------lookups",lookups);
  // console.log("CriminalaData-------MatchedUserInfo",MatchedUserInfo);
  // console.log("CriminalaData-------data",data);
  // console.log("CriminalaData-------rowId",rowId);
  // let latentCrimeType =
  //   lookups &&
  //   lookups
  //     .find(data => data.lookupName === "LatentCrimeType")
  //     .items.map(item => {
  //       return {
  //         displayName: item.englishText,
  //         value: item.englishText
  //       };
  //     });
  const lang = localStorage.getItem("lang") || "en";

  const latentCrimeType = [];

  const crimeCityOptions = lookups && Array.isArray(lookups) && lookups.find(data => data.lookupName === "LatentCrimeCity").items.map(item => {
    return {
      displayName: lang === "en" ? item.englishText : item.arabicText,
      value: item.id
    };
  });
  // console.log("crimeCityOptions",crimeCityOptions,lang);
  // console.log("crimeCityOptionsLookup",lookups);





  const crimeRegionOptions = lookups && Array.isArray(lookups) && lookups.find(data => data.lookupName === "LatentCrimeRegion").items.map(item => {
    return {
      displayName: lang === "en" ? item.englishText : item.arabicText,
      value: item.id
    };
  });
  // console.log("crimeRegionOptions",crimeRegionOptions);

  // const crimeTypeOptions = lookups.find(data => data.lookupName === "LatentCrimeType").items.map( item => {
  //   return {
  //     displayName: lang === "en" ? item.englishText : item.arabicText,
  //     value: item.id
  //   };
  // });


  // console.log("crimeTypes",crimeTypes);

  const codeOptions = crimeTypes.lookups.map(item => {
    return {
      displayName: `${item.eventDesc} - ${item.classDesc}`,
      value: `${item.eventCode}|${item.classCode}`,
    };
  });
  // console.log("codeOptions",crimeTypes.lookups);




  if (!crimeCityOptions) {
    requestShowNotification({
      message: formatMessage({ id: 'lookupNotFoundCrimeCity' }),
      type: "is-danger"
    });
  }
  if (!crimeRegionOptions) {
    requestShowNotification({
      message: formatMessage({ id: 'lookupNotFoundCrimeRegion' }),
      type: "is-danger"
    });
  }
  if (!codeOptions) {
    requestShowNotification({
      message: formatMessage({ id: 'lookupNotFoundEventClassCode' }),
      type: "is-danger"
    });
  }





  if (rowId && rowId === undefined || rowId === "abcd") {
    MatchedUserInfo = ""
  }
  return (
    <Fragment>
      <div className="crime-data">
        <div className="columns">
          <div className="column">
            <div className="control">
              <Label
                htmlFor="bioname"
                text={formatMessage({ id: "crimePlace" })}
              />
              <Input
                type="text"
                className="input"
                id="crimePlace"
                maxLength="25"
                onChange={e => {
                  handleInputChange(e, "crimePlace");
                }}
                placeholder={formatMessage({ id: "crimePlace" })}
                value={
                  data && data.crimePlace
                    ? data.crimePlace
                    : data.crimePlace === ""
                      ? ""
                      : MatchedUserInfo.person &&
                        MatchedUserInfo.person.extendedData
                        ? Array.isArray(MatchedUserInfo.person.extendedData) && MatchedUserInfo.person.extendedData.find(
                          data => data.key === "Crime Place"
                        )
                          ? Array.isArray(MatchedUserInfo.person.extendedData) && MatchedUserInfo.person.extendedData.find(
                            data => data.key === "Crime Place"
                          ).value
                          : ""
                        : ""
                }
              />
            </div>
          </div>


          {/*
          <div className="column">
            <div className="control">
              <Label
                htmlFor="crimeType"
                text={formatMessage({ id: "crimeType" })}
              />
              <Select
                defaultValue={
                  MatchedUserInfo.person && MatchedUserInfo.person.extendedData
                    ? MatchedUserInfo.person.extendedData.find(
                        data => data.key === "Crime Type"
                      )
                      ? MatchedUserInfo.person.extendedData.find(
                          data => data.key === "Crime Type"
                        ).value
                      : formatMessage({ id: "crimeType" })
                    : formatMessage({ id: "crimeType" })
                }
                className={"level-right "}
                formatMessage={formatMessage}
                name={formatMessage({ id: "crimeType" })}
                options={crimeTypeOptions}
                onChange={e => {
                  handleInputChange(e, "crimeType");
                }}
              />
            </div>
          </div>
          */}



          <div className="column">
            <div className="control">
              <Label
                htmlFor="eventCode"
                text={formatMessage({ id: "eventClassCode" })}
              />
              <Select
                className={"level-right "}
                formatMessage={formatMessage}
                name={formatMessage({ id: "eventClassCode" })}
                options={codeOptions}
                defaultValue={`${data.eventCode}|${data.classCode}`}
                onChange={e => {
                  handleInputChange(e, "eventClassCode");
                }}
              />
            </div>
          </div>


          {/*
          <div className="column">
            <div className="control">
              <Label
                htmlFor="classCode"
                text={formatMessage({ id: "classCode" })}
              />
              <Select
                defaultValue={
                  MatchedUserInfo.person && MatchedUserInfo.person.extendedData
                    ? MatchedUserInfo.person.extendedData.find(
                        data => data.key === "Class Code"
                      )
                      ? MatchedUserInfo.person.extendedData.find(
                          data => data.key === "Class Code"
                        ).value
                      : formatMessage({ id: "classCode" })
                    : formatMessage({ id: "classCode" })
                }
                className={"level-right "}
                formatMessage={formatMessage}
                name={formatMessage({ id: "classCode" })}
                options={ codeOptions }
                onChange={e => {
                  handleInputChange(e, "classCode");
                }}
              />
            </div>
          </div>
          */}





          <div className="column">
            <div className="control">
              <Label
                htmlFor="crimeRegion"
                text={formatMessage({ id: "crimeRegion" })}
              />
              <Select
                defaultValue={data.crimeRegion}
                className={"level-right "}
                formatMessage={formatMessage}
                name={formatMessage({ id: "crimeRegion" })}
                options={crimeRegionOptions}
                onChange={e => {
                  handleInputChange(e, "crimeRegion");
                }}
              />
            </div>
          </div>

          <div className="column">
            <div className="control">
              <Label
                htmlFor="crimeCity"
                text={formatMessage({ id: "crimeCity" })}
              />
              <Select
                defaultValue={data.crimeCity}
                className={"level-right "}
                formatMessage={formatMessage}
                name={formatMessage({ id: "crimeCity" })}
                options={crimeCityOptions}
                onChange={e => {
                  handleInputChange(e, "crimeCity");
                }}
              />
            </div>
          </div>

        </div>
        <div className="columns">
          <div className="column">
            <div className="control">
              <Label
                htmlFor="crimeDate"
                text={formatMessage({ id: "crimeDate" })}
              />

              <HijriDatePicker
                className="input"
                id="crimeDate"
                placeholder={formatMessage({ id: "crimeDate" })}
                value={
                  (data.crimeDate ? data.crimeDate :
                    MatchedUserInfo && MatchedUserInfo.person && MatchedUserInfo.person.extendedData && Array.isArray(MatchedUserInfo.person.extendedData) &&
                      MatchedUserInfo.person.extendedData.find(
                        data => data.key === "Date of Crime"
                      )
                      ? MatchedUserInfo && MatchedUserInfo.person && Array.isArray(MatchedUserInfo.person.extendedData) && MatchedUserInfo.person.extendedData.find(
                        data => data.key === "Date of Crime"
                      ).value
                      : null
                  )
                }
                // value="15/10/1433"
                onChange={e => {
                  // handleInputChange(e, "crimeDate");
                  handleInputChange({ target: { value: e } }, "crimeDate");
                }}
              />
            </div>
          </div>
          <div className="column is-two-fifths">
            <div className="control">
              <Label
                htmlFor="caseInvestigationDepartment"
                text={formatMessage({ id: "caseInvestigationDepartment" })}
              />
              <Input
                type="text"
                className="input"
                id="caseInvestigationDepartment"
                // maxLength="10"
                onChange={e => {
                  handleInputChange(e, "caseInvestigationDepartment");
                }}
                placeholder={formatMessage({ id: "caseInvestigationDepartment" })}
                value={
                  data.caseInvestigationDepartment
                    ? data.caseInvestigationDepartment
                    : data.caseInvestigationDepartment === ""
                      ? ""
                      : MatchedUserInfo.person &&
                        MatchedUserInfo.person.extendedData
                        ? Array.isArray(MatchedUserInfo.person.extendedData) && MatchedUserInfo.person.extendedData.find(
                          data => data.key === "Investigation Department"
                        )
                          ? Array.isArray(MatchedUserInfo.person.extendedData) && MatchedUserInfo.person.extendedData.find(
                            data => data.key === "Investigation Department"
                          ).value
                          : ""
                        : ""
                }
              />
            </div>
          </div>
          <div className="column">
            <div className="control">
              <Label
                htmlFor="casefilenumber"
                text={formatMessage({
                  id: "casefilenumber" //Case File Number
                })}
              />
              <Input
                type="text"
                className="input"
                id="casefilenumber"
                maxLength="10"
                onChange={e => {
                  handleInputChange(e, "caseFileNumber");
                }}
                placeholder={formatMessage({ id: "casefilenumber" })}
                value={
                  data.caseFileNumber
                    ? data.caseFileNumber
                    : data.caseFileNumber === ""
                      ? ""
                      : MatchedUserInfo.person &&
                        MatchedUserInfo.person.extendedData
                        ? Array.isArray(MatchedUserInfo.person.extendedData) && MatchedUserInfo.person.extendedData.find(
                          data => data.key === "Case File Number"
                        )
                          ? Array.isArray(MatchedUserInfo.person.extendedData) && MatchedUserInfo.person.extendedData.find(
                            data => data.key === "Case File Number"
                          ).value
                          : ""
                        : ""
                }
              />
            </div>
          </div>
        </div>
        <div className="columns">
          <div className="column">
            <div className="control">
              <Label
                htmlFor="crimeNoteLong"
                text={formatMessage({ id: "crimeNoteLong" })}
              />
              <Input
                type="text"
                className="input"
                id="crimeNoteLong"
                maxLength="50"
                onChange={e => {
                  handleInputChange(e, "crimeNoteLong");
                }}
                placeholder={formatMessage({ id: "crimeNoteLong" })}
                value={
                  data.crimeNoteLong
                    ? data.crimeNoteLong
                    : data.crimeNoteLong === ""
                      ? ""
                      : MatchedUserInfo.person &&
                        MatchedUserInfo.person.extendedData
                        ? Array.isArray(MatchedUserInfo.person.extendedData) && MatchedUserInfo.person.extendedData.find(
                          data => data.key === "Note"
                        )
                          ? Array.isArray(MatchedUserInfo.person.extendedData) && MatchedUserInfo.person.extendedData.find(
                            data => data.key === "Note"
                          ).value
                          : ""
                        : ""
                }
              />
            </div>
          </div>
        </div>
        <div className="columns">
          <div className="space-fill" />
        </div>
      </div>
    </Fragment>
  );
}
// export default CriminalaData;
export default connect(
  () => ({}),
  {
    requestShowNotification
  }
)(CriminalaData);
