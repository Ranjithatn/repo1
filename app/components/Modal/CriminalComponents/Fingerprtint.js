import React, { Fragment } from "react";
import { connect } from "react-redux";
import { Link, withRouter } from "react-router-dom";
import NumericInput from "react-numeric-input";
import Modal from "../Modal";
import Label from "../../Label/Label";
import Image from "../../Image/Image";
import { reg_space } from "../../../utils/regEx";
// import "./UpdateCriminalModal.scss";

import { sortFingers } from "../../Tenprint/TenprintVerify/Helper";

function Fingerprtint({ formatMessage, JobType, LeftFingers, RightFingers }) {
  // console.log("Fingerprtint :: LeftFingers, RightFingers",LeftFingers, RightFingers);

  let sortedLeftFingers = LeftFingers;
  let sortedRightFingers = RightFingers;

  if (
    sortedLeftFingers &&
    sortedLeftFingers.fingers &&
    sortedLeftFingers.fingers.length > 0
  ) {
    sortedLeftFingers = sortFingers(LeftFingers);
  }

  if (
    sortedRightFingers &&
    sortedRightFingers.fingers &&
    sortedRightFingers.fingers.length > 0
  ) {
    sortedRightFingers = sortFingers(RightFingers);
  }

  // if ( sortedLeftFingers.fingers.length === 0 && LeftFingers.fingers.length > 0 ) { sortedLeftFingers = LeftFingers; }
  // if ( sortedRightFingers.fingers.length === 0 && RightFingers.fingers.length > 0 ) { sortedRightFingers = RightFingers; }

  // console.log("sortedLeftFingers",sortedLeftFingers);
  // console.log("sortedRightFingers",sortedRightFingers);

  return (
    <Fragment>
      {JobType !== "Custom" ? (
        <div className="tenprint-fields">
          <Label text={formatMessage({ id: "fingerprints" })} />
          <div className="columns">
            {sortedLeftFingers && sortedLeftFingers.length > 0 ? (
              <div className="column">
                {sortedLeftFingers.map((name, i) => {
                  return (
                    <Fragment>
                      <div
                        key={name + i}
                        className="segmented-data-container"
                        title={
                          name.annotation
                            ? name.annotation + ":" + name.annotationNote
                            : ""
                        }
                      >
                        {name.thumbNail !== "" ? (
                          <Image
                            src={"data:image/png;base64," + name.thumbNail}
                            id={name.position}
                          />
                        ) : (
                          <span title={name.annotationNote}>
                            {name.annotation}
                          </span>
                        )}
                      </div>
                      <span className="impression-name">
                        {name.position === "LeftRing" &&
                        JobType === "Latent" ? (
                          <span>{formatMessage({ id: "Unknown" })}</span>
                        ) : (
                          formatMessage({ id: name.position })
                        )}
                      </span>
                      {/* <Switch
                        id={name.position + i}
                        value={name.position}
                        className="is-primary is-rounded"
                      /> */}
                    </Fragment>
                  );
                })}
              </div>
            ) : (
              ""
            )}
            {sortedRightFingers && sortedRightFingers.length > 0 ? (
              <div className="column ">
                {sortedRightFingers.map((name, i) => {
                  return (
                    <Fragment>
                      <div
                        className="segmented-data-container"
                        title={
                          name.annotation
                            ? name.annotation + ":" + name.annotationNote
                            : ""
                        }
                      >
                        {name.thumbNail !== "" ? (
                          <Image
                            src={"data:image/png;base64," + name.thumbNail}
                            id={name.position}
                          />
                        ) : (
                          <span title={name.annotationNote}>
                            {name.annotation}s
                          </span>
                        )}
                      </div>

                      <span className="impression-name">
                        {formatMessage({ id: name.position })}
                      </span>
                      {/* <Switch
                        id={name.position + i}
                        value={name.position}
                        className="is-primary is-rounded"
                      /> */}
                    </Fragment>
                  );
                })}
              </div>
            ) : (
              ""
            )}
          </div>
          {/* <span className="noteLabel">
            {" "}
            {formatMessage({ id: "registerNote" })}{" "}
          </span> */}
        </div>
      ) : (
        ""
      )}
    </Fragment>
  );
}
export default Fingerprtint;
