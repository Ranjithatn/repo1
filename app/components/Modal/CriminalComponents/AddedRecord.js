import React, { Fragment } from "react";
import { connect } from "react-redux";
import { Link, withRouter } from "react-router-dom";
import Modal from "../Modal";
import Button from "../../Button/Button";
import { requestHideModal } from "../../../actions/modal";
import { requestInputFieldChanged } from "../../../actions/global";
import Label from "../../Label/Label";
import Select from "../../Select/Select";
import Input from "../../Input/Input";
import Icon from "../../Icon/Icon";
import { requestSetAddnew } from "../../../actions/tenprint";
import CriminalHistory from "./CriminalHistory";

// import "./UpdateCriminalModal.scss";

export function AddedRecord({ formatMessage, presentObj, onClick, index,criminalHistory }) {
  const handleChange = e => {
    console.log("handle click");
  };

  return (
    <div className="case-field">
      <div className="control">
        <Input
          type="text"
          className="input"
          id="lashing"
          // disabled={!lashing_checkbox}
          onChange={e => {
            handleChange(e);
          }}
          value={presentObj.CaseType}
          disabled={ criminalHistory ? true : false }
        />
      </div>
      <div className="control">
        <Input
          type="text"
          className="input"
          id="crimeDesc"
          placeholder={formatMessage({
            id: "typeSubject"
          })}
          onChange={e => {
            handleChange(e);
          }}
          value={presentObj.Description}
          disabled={ criminalHistory ? true : false }
        />
      </div>
      <div className="icon-div">
      <Icon
          icon="trash"
          onClick={ criminalHistory ? () => {} : () => onClick(index)}
          title={ criminalHistory ? "" : formatMessage({ id: "deleteJob" })}
          disabled={ criminalHistory ? true : false }
        />

      </div>
    </div>
  );
}
export default connect(null, { requestInputFieldChanged, requestSetAddnew })(
  AddedRecord
);
