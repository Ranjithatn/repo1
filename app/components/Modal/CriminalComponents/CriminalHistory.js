import React, { Fragment } from "react";
import { connect } from "react-redux";
import { Link, withRouter } from "react-router-dom";
import Modal from "../Modal";
import Button from "../../Button/Button";
import { requestHideModal } from "../../../actions/modal";
import Label from "../../Label/Label";
import AddedRecord from "./AddedRecord";
import Punishment from "./Punishment";

// import "./UpdateCriminalModal.scss";

function CriminalHistory({
  formatMessage,
  MatchedUserInfo,
  handleCheckboxClick,
  lashing_checkbox,
  fine_checkbox,
  handleInputClick,
  handleInputChange,
  data
}) {
  return (
    <Fragment>
      {MatchedUserInfo &&
        MatchedUserInfo.map((obj, index) => {
          const criminalclass = obj.additionalMatchDatas.find( data => data.key === "Criminal Class" );
          return (
            <div className="criminal-fields" key={index}>
              <Label
                text={
                  formatMessage({ id: "recordHistory" }) + "  " + (index + 1)
                }
              />
              <div className="case-content">
                {obj.additionalMatchDatas&&obj.additionalMatchDatas.length > 0 ? (
                  <div className="case-field">
                    <span className="span-head">
                      {formatMessage({ id: "caseType" })}
                    </span>
                    <span className="span-head">
                      {formatMessage({ id: "desc" })}
                    </span>
                    <span />
                  </div>
                ) : (
                  ""
                )}

                {obj.additionalMatchDatas&&obj.additionalMatchDatas.length > 0 && criminalclass && criminalclass.value &&
                  criminalclass && criminalclass.value.map((presentObj, index) => {
                   
                    return (
                      <Fragment key={index}>
                        <AddedRecord
                          formatMessage={formatMessage}
                          presentObj={presentObj}
                          criminalHistory={true}
                        />
                      </Fragment>
                    );
                  })}
              </div>
              {/* {data.push({["lashing"+index]:""})} */}
              <Fragment key={index}>
                <Punishment
                  formatMessage={formatMessage}
                  MatchedUserInfo={obj.additionalMatchDatas?obj.additionalMatchDatas:{}}
                  handleCheckboxClick={handleCheckboxClick}
                  lashing_checkbox={lashing_checkbox}
                  fine_checkbox={fine_checkbox}
                  data={data}
                  index={index}
                  handleInputClick={handleInputClick}
                  handleInputChange={handleInputChange}
                  criminalHistory={true}
                />
              </Fragment>
            </div>
          );
        })}
    </Fragment>
  );
}
export default CriminalHistory;