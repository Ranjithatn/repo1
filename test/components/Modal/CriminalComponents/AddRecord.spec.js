import React from "react";
import toJson from "enzyme-to-json";
import { AddRecord } from "../../../../app/components/Modal/CriminalComponents/AddRecord";
import Adapter from "enzyme-adapter-react-16";
import { configure, shallow, mount } from "enzyme";


configure({ adapter: new Adapter() });

const props = {
  formatMessage: data => data.id,
  newCrimeType: { CaseType: "" }
}

describe('<AddRecord />', () => {

  it("should render without crashing", () => {
    const component = mount(
      <AddRecord {...props}/>
    );
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });
});
