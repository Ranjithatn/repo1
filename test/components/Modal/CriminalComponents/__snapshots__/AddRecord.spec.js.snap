// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddRecord /> should render without crashing 1`] = `
<AddRecord
  formatMessage={[Function]}
  newCrimeType={
    Object {
      "CaseType": "",
    }
  }
>
  <div
    className="case-field"
  >
    <div
      className="control"
    >
      <Select
        defaultValue="Filter"
        formatMessage={[Function]}
        id="CaseType"
        name="CrimeType"
        onChange={[Function]}
        options={
          Array [
            Object {
              "displayName": "Drug",
              "value": "Drug",
            },
            Object {
              "displayName": "Murder",
              "value": "Murder",
            },
            Object {
              "displayName": "Robbery",
              "value": "Robbery",
            },
          ]
        }
        value=""
      >
        <div
          className="select "
        >
          <select
            defaultValue="Filter"
            disabled={false}
            id="CaseType"
            name="CrimeType"
            onChange={[Function]}
            style={Object {}}
            value=""
          >
            <option
              selected={true}
              value=""
            />
            <option
              key="Drug"
              value="Drug"
            >
               
              Drug
            </option>
            <option
              key="Murder"
              value="Murder"
            >
               
              Murder
            </option>
            <option
              key="Robbery"
              value="Robbery"
            >
               
              Robbery
            </option>
          </select>
        </div>
      </Select>
    </div>
    <div
      className="control"
    >
      <Input
        className="input"
        id="Description"
        onChange={[Function]}
        placeholder="typeSubject"
        type="text"
      >
        <div
          className="control"
          style={Object {}}
        >
          <input
            className="input"
            disabled={false}
            id="Description"
            onChange={[Function]}
            onKeyPress={[Function]}
            placeholder="typeSubject"
            style={Object {}}
            type="text"
          />
        </div>
      </Input>
    </div>
    <Button
      className="is-primary is-rounded"
      disabled={true}
      id="upmodbackbtn"
      leftIcon="plus"
      onClick={[Function]}
    >
      <button
        className="button is-primary is-rounded"
        disabled={true}
        id="upmodbackbtn"
        onClick={[Function]}
        style={Object {}}
        type="button"
      >
        <Icon
          icon="plus"
        >
          <span
            className="icon"
            disabled={false}
            onClick={[Function]}
            style={Object {}}
            title=""
          >
            <i
              className="fa fa-plus"
            />
          </span>
        </Icon>
      </button>
    </Button>
  </div>
</AddRecord>
`;
